File: 백준/Silver/2942. 퍼거슨과 사과/퍼거슨과 사과.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int R = Integer.parseInt(st.nextToken());
        int G = Integer.parseInt(st.nextToken());

        int tmp = gcd(R, G);

        List<Integer> answer = new ArrayList<>();
        for (int i = 1; i <= (int) Math.sqrt(tmp); i++) {
            if (tmp % i == 0) {
                answer.add(i);
                if (tmp / i == i) {
                    continue;
                }
                answer.add(tmp / i);
            }
        }

        for (int i = 0; i < answer.size(); i++) {
            int num = answer.get(i);
            System.out.println(num + " " + (R / num) + " " + (G / num));
        }
    }

    static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        } else {
            return gcd(b, a % b);
        }
    }
}

File: 백준/Silver/2942. 퍼거슨과 사과/퍼거슨과 사과.py
from math import sqrt
R, G = map(int, input().split())

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

tmp = gcd(R, G)
ans = []

for i in range(1, int(sqrt(tmp)) + 1):
    if tmp % i == 0:
        ans.append(i)
        if tmp // i == i:
            continue
        ans.append(tmp // i)

for i in ans:
    print(i, R//i, G//i)
